### PR TITLE
Add withholding taxes to invoices info

### PR DESCRIPTION
**Pull request by luisaterrabrito**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -2039,6 +2039,8 @@ Get details for a single invoice.
                                 + tax_exclusive_before_discount (Money)
                                 + tax_inclusive (Money)
                                 + tax_inclusive_before_discount (Money)
+                            + withholding_tax (object)
```
Comment (dcsg) [inline apiary.apib]: Q/B: this should be nullable, right?

File: apiary.apib [hunk 2]
```diff
@@ -2049,6 +2051,11 @@ Get details for a single invoice.
                         + rate: 0.21 (number)
                         + taxable (Money)
                         + tax (Money)
+                + withholding_taxes (array)
```
Comment (dcsg) [inline apiary.apib]: Will be weird for the countries where it does not apply to have this in the response.

**Review by dcsg — CHANGES_REQUESTED** [review]

**Review by dcsg — DISMISSED** [review]

**Comment by luisaterrabrito** [comment]
Actually, implementing this, I noticed the vat rate is presented as zero not null. The withholding tax might not apply to an invoice in case the country is not Italy, but we cannot check that for an item. So should the withholding tax return the tax as zero either way?
@dcsg @jenssegers @steftrenson @jdrieghe

**Comment by dcsg** [comment]
IMO if the user didn't set the withholding tax than it should not be zero and the object should not exist or the `withholding_tax` is null

**Review by tiagobastos — APPROVED** [review]

**Review by steftrenson — APPROVED** [review]

**Review by dcsg — APPROVED** [review]

**Pull request merged**